ql/src/java/org/apache/hadoop/hive/ql/optimizer/ParallelEdgeFixer.java

```diff
@@ -239,4 +251,18 @@ private String extractColumnName(ExprNodeDesc expr) {
     }
     throw new RuntimeException("unexpected mapping expression!");
```

Comment:
nit: a more specific checked exception would suit better here or just return null a later in `colMappingInverseKeys` handle the specific exception or null only.